apps/dsaparam.c

```diff
@@ -249,11 +282,21 @@ int dsaparam_main(int argc, char **argv)
     if (genkey) {
         DSA *dsakey;
 
-        if ((dsakey = DSAparams_dup(dsa)) == NULL)
+        EVP_PKEY_CTX_free(ctx);
```

Comment:
This does not look correct.

Reply:
Yeah, it's broken.  Keygen is required to finish this PR, I jumped the gun a bit.
Back on hold.